LICENSE.txt

```diff
@@ -2290,3 +2290,46 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
+
+--------------------------------------------------------------------------------
+The files cpp/src/arrow/vendored/whereami/whereami.h, 
+cpp/src/arrow/vendored/whereami/whereami.cc are adapted from
+Grégory Pakosz's whereami library (https://github.com/gpakosz/whereami)
+It is dual licensed under both the WTFPLv2 and MIT licenses.
```

Comment:
Reminder to myself to post in community PR:

Copying both WTFPLv2 and MIT licenses from https://github.com/gpakosz/whereami/commit/e64e8b379bdabffcb121e9d6993d786c5e19c349